src/TiogaBlock.cpp

```diff
@@ -175,6 +174,7 @@ TiogaBlock::update_iblanks()
   auto& iblarr = bdata_.iblank_.d_view;
   auto& nidmap = bdata_.eid_map_.d_view;
   auto& iblank_ngp = stk::mesh::get_updated_ngp_field<double>(*ibf);
+  ibf->sync_to_device();
```

Comment:
Is this required? This doesn't make sense to me for two reasons - 
1. If it is indeed required, this should happen before line 176
2. The loop below is actually changing ibf, so syncing to device shouldn't really matter here.

Reply:
If this is not called, the code fails and displays the "Error: Modify on device called for Field: iblank but it has an uncleared modified_on_host" error. Calling sync_to_device() also calls reset_execution_space() inside STK, which I think resets the m_modified_on_host variable to eliminate the error. There might be a better way to do this.

Reply:
Moved ibf->sync_to_device() to right after the ibf definition, and updated the fork.